data-prepper-core/src/main/java/com/amazon/dataprepper/plugin/DefaultPluginFactory.java

```diff
@@ -30,17 +31,21 @@
     private final Collection<PluginProvider> pluginProviders;
     private final PluginCreator pluginCreator;
     private final PluginConfigurationConverter pluginConfigurationConverter;
+    private final ApplicationContext applicationContext;
```

Comment:
You can use a `BeanFactory` instead of `ApplicationContext` here and throughout this PR. It is a lower-level interface which provides what you need. Aside from that change, I think the approach of providing the `BeanFactory` here is great.